src/app/qgsdxfexportdialog.cpp

```diff
@@ -800,6 +808,239 @@ void QgsDxfExportDialog::deselectDataDefinedBlocks()
+  if ( !myDirInfo.isWritable() )
+  {
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "The directory containing your dataset needs to be writable!" ) );
+    return;
+  }
+
+  QFile myFile( outputFileName );
+  if ( myFile.open( QFile::WriteOnly | QFile::Truncate ) )
+  {
+    QTextStream myFileStream( &myFile );
+    // save as utf-8 with 2 spaces for indents
+    myDocument.save( myFileStream, 2 );
+    myFile.close();
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "Created DXF settings file as %1" ).arg( outputFileName ) );
+    settings.setValue( QStringLiteral( "dxf/lastSettingsDir" ), QFileInfo( outputFileName ).absolutePath() );
+    return;
+  }
+  else
+  {
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "ERROR: Failed to created DXF Export settings file as %1. Check file permissions and retry." ).arg( outputFileName ) );
+    return;
+  }
+}
+
+
+void QgsDxfExportDialog::saveSettingsToXML( QDomDocument &doc ) const
+{
+  QDomImplementation DomImplementation;
+  const QDomDocumentType documentType = DomImplementation.createDocumentType( QStringLiteral( "qgis" ), QStringLiteral( "http://mrcc.com/qgis.dtd" ), QStringLiteral( "SYSTEM" ) );
+  QDomDocument myDocument( documentType );
+
+  QDomElement myRootNode = myDocument.createElement( QStringLiteral( "qgis" ) );
+  myRootNode.setAttribute( QStringLiteral( "version" ), Qgis::version() );
+  myDocument.appendChild( myRootNode );
+
+  QDomElement symbologyModeElement = myDocument.createElement( QStringLiteral( "symbology_mode" ) );
+  symbologyModeElement.appendChild( QgsXmlUtils::writeVariant( static_cast<int>( symbologyMode() ), doc ) );
+  myRootNode.appendChild( symbologyModeElement );
+
+  QDomElement symbologyScaleElement = myDocument.createElement( QStringLiteral( "symbology_scale" ) );
+  symbologyScaleElement.appendChild( QgsXmlUtils::writeVariant( symbologyScale(), doc ) );
+  myRootNode.appendChild( symbologyScaleElement );
+
+  QDomElement encodingElement = myDocument.createElement( QStringLiteral( "encoding" ) );
+  encodingElement.appendChild( QgsXmlUtils::writeVariant( encoding(), doc ) );
+  myRootNode.appendChild( encodingElement );
+
+  QDomElement crsElement = myDocument.createElement( QStringLiteral( "crs" ) );
+  crsElement.appendChild( QgsXmlUtils::writeVariant( crs(), doc ) );
+  myRootNode.appendChild( crsElement );
+
+  QDomElement mapThemeElement = myDocument.createElement( QStringLiteral( "map_theme" ) );
+  mapThemeElement.appendChild( QgsXmlUtils::writeVariant( mapTheme(), doc ) );
+  myRootNode.appendChild( mapThemeElement );
+
+  QDomElement layersElement = myDocument.createElement( QStringLiteral( "layers" ) );
+  for ( const auto dxfLayer : layers() )
+  {
+    QgsVectorLayer *vl = dxfLayer.layer();
+    QDomElement layerElement = myDocument.createElement( QStringLiteral( "layer" ) );
+    layerElement.setAttribute( QStringLiteral( "source" ), vl->publicSource() );
```

Comment:
Only saving the source can be problematic (e.g., the same source could be representing two distinct vector layers with different symbologies).

I'd use the QgsVectorLayerRef class and rely on its writeXml / readXml + resolve functions.

Reply:
Addressed in https://github.com/gacarrillor/QGIS/pull/37/commits/df06cbef638b4649aafe9964b66a9ae73f2e36b4